service/src/main/java/org/apache/griffin/core/util/MeasureUtil.java

```diff
@@ -54,6 +55,27 @@ private static void validateGriffinMeasure(GriffinMeasure measure) {
             throw new GriffinException.BadRequestException
                     (INVALID_CONNECTOR_NAME);
         }
+        if (!validatePredicates(measure)) {
+            throw new GriffinException.BadRequestException(INVALID_MEASURE_PREDICATE);
+        }
+    }
+
+    private static boolean validatePredicates(GriffinMeasure measure) {
+        for (DataSource dataSource : measure.getDataSources()) {
+            for (DataConnector dataConnector: dataSource.getConnectors()) {
+                for (SegmentPredicate segmentPredicate : dataConnector.getPredicates()) {
+                    try {
+                        Predicator predicator = PredicatorFactory.newPredicateInstance(segmentPredicate);
+                        if (predicator == null) {
```

Comment:
we throw NotFoundException, so this is not needed

Reply:
fixed